### PR TITLE
Doxygen style guide

### PR DESCRIPTION
**Pull request by HaoZeke**
----
This PR is part of the **Google Season of Docs 2020**

### CODE REVIEW

**Comment by HaoZeke** [comment]
@certik and @isuruf this is now much more readable, and should be good to go.

File: docs/Doxygen/md/doc_guide.md [hunk 1]
```diff
@@ -1,9 +1,214 @@
-# API Documentation Guide
+# Doxygen Style Guide
 
-We write all the API documentation with [Doxygen](https://doxygen.nl). Note the following conventions:
+We write all the API documentation with [Doxygen](https://doxygen.nl). The
+`add.h` and `add.cpp` are reference implementations of the style guide.
+
+## Language and Locale
+
+- Documentation should be in english
```
Comment (isuruf) [inline docs/Doxygen/md/doc_guide.md:8]:
```suggestion
- Documentation should be in English
```

File: docs/Doxygen/md/doc_guide.md [hunk 2]
```diff
@@ -1,9 +1,214 @@
...
+- All dates are `YYYY-MM-DD`
+- We strongly discourage the use of unicode symbols
```
Comment (isuruf) [inline docs/Doxygen/md/doc_guide.md:10]:
```suggestion
- We strongly discourage the use of Unicode symbols
```

File: docs/Doxygen/md/doc_guide.md [hunk 3]
```diff
@@ -1,9 +1,214 @@
...
+Classes descriptions have the `@class` directive declared before `@brief` in `.h` files and
+before `@details` in the `.cpp` as well.
+
+### Functions
+
+- All documentation blocks must precede the function being documented
+
+## External Tools
+
+We expect the build system to be able to find `graphiz` for `dot`.
```
Comment (isuruf) [inline docs/Doxygen/md/doc_guide.md:212]:
```suggestion
We expect the build system to be able to find `graphviz` for `dot`.
```

File: docs/Doxygen/md/doc_guide.md [hunk 4]
```diff
@@ -1,9 +1,214 @@
...
-    - Also `/**` for comment blocks
-    - Use `//!` for inline details and  `///` for brief inline descriptions
 - We **do** expect private variables to be documented
+
+Classes descriptions have the `@class` directive declared before `@brief` in `.h` files and
+before `@details` in the `.cpp` as well.
+
+### Functions
+
+- All documentation blocks must precede the function being documented
```
Comment (isuruf) [inline docs/Doxygen/md/doc_guide.md:208]: Use `.`s at the end of all sentences

**Comment by HaoZeke** [comment]
Thanks for the quick and useful review @isuruf !

**Comment by HaoZeke** [comment]
@certik is this good to go?

**Pull request merged**